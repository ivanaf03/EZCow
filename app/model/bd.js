import * as SQLite from "expo-sqlite";

let db;

export const getDatabase = async () => {
  if (!db) {
    db = await SQLite.openDatabaseAsync("ezcow.db");
  }
  return db;
};

export const initDatabase = async () => {
  db = await getDatabase();

  await db.execAsync(`
        PRAGMA journal_mode = WAL;

        CREATE TABLE IF NOT EXISTS User (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            email TEXT NOT NULL UNIQUE,
            password TEXT DEFAULT NULL,
            google_id TEXT DEFAULT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            CHECK (password IS NOT NULL OR google_id IS NOT NULL)
        );

        CREATE TABLE IF NOT EXISTS Cow (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            code TEXT NOT NULL UNIQUE,
            name TEXT NOT NULL,
            entryDate DATE NOT NULL,
            exitDate DATE DEFAULT NULL,
            gender TEXT NOT NULL,
            breed TEXT NOT NULL,
            phase TEXT NOT NULL,
            user_fk INTEGER NOT NULL,
            mother_fk INTEGER DEFAULT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id),
            FOREIGN KEY(mother_fk) REFERENCES Cow(id)
        );

        CREATE TABLE IF NOT EXISTS Field (
            id INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            latitude REAL DEFAULT 43.0,
            longitude REAL DEFAULT -7.0,
            user_fk INTEGER NOT NULL,
            FOREIGN KEY(user_fk) REFERENCES User(id)
        );

        CREATE TABLE IF NOT EXISTS BreedingEvent (
            if INTEGER PRIMARY KEY AUTOINCREMENT,
            name TEXT NOT NULL,
            description TEXT,
            date DATE NOT NULL,
            cow_fk INTEGER NOT NULL,
            FOREIGN KEY(cow_fk) REFERENCES Cow(id)
        );
    `);
};

export const deleteDatabase = async () => {
  await db.closeAsync();
  await SQLite.deleteDatabaseAsync("ezcow.db");
};

export const closeDatabase = async () => {
  await db.closeAsync();
  db = null;
};

export const syncDatabase = async () => {
  await db.execAsync(`
        PRAGMA wal_checkpoint(FULL);
    `);
};
