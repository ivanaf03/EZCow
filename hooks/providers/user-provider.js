import React from 'react';

const UserContext = React.createContext();

const UserProvider = ({ children }) => {
    const [user, setUser] = React.useState(null);

    const login = (data) => {
        setUser(data);
    };

    const logout = () => {
        setUser(null);
    };

    return (
        <UserContext.Provider value={{ user, login, logout }}>
            {children}
        </UserContext.Provider>
    );  
};

export default UserProvider;

export const useUser = () => React.useContext(UserContext);