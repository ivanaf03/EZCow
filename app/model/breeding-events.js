import { getDatabase } from "./bd";

export const insertBreedingEvent = async (
  cow_fk,
  description,
  date,
  user_fk
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, description, date, user_fk) VALUES (?, ?, ?, ?)`,
      [cow_fk, description, date, user_fk]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getBreedingEventsByDayAnUserId = async (day, userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE date = ? AND user_fk = ?`,
      [userId, day]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};