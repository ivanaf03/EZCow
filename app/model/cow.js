import { getDatabase } from './bd';

export const insertCow = async (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) => {

    db = await getDatabase();

    const res = await db.runAsync(
        `INSERT INTO Cow (code, name, entryDate, gender, breed, phase, user_fk, mother_fk) VALUES (?, ?, ?, ?, ?, ?, ?, ?)`,
        [code, name, entryDate, gender, breed, phase, user_fk, mother_fk]
    );

    return res;
};

export const getCowIdByCode = async (code) => {

    db = await getDatabase();

    const res = await db.getFirstAsync(
        `SELECT id FROM Cow WHERE code = ?`,
        [code]
    );

    return res.id;
};

export const getAvailableCowNameById = async (id) => {

    db = await getDatabase();

    const res = await db.getFirstAsync(
        `SELECT name FROM Cow WHERE id = ? AND exitDate is NULL`,
        [id]
    );

    return res.name;
};

export const getAvailableCowCodeByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT code FROM Cow WHERE user_fk = ? AND exitDate is NULL AND gender='Femenino' AND phase='Adulto'`,
        [userId]
    );

    return res.map((cow) => cow.code);
};

export const getAllCowsAvailableByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL`,
        [userId]
    );

    return res;
};

export const getAllCowsAvailableByUserIdAndPhaseAndGender = async (userId, phase, gender) => {
    
    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = ? AND gender = ?`,
        [userId, phase, gender]
    );

    return res;
};

export const getCalvesAvailableByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NULL AND phase = 'Ternero'`,
        [userId]
    );

    return res;
};

export const setExitDateByCowId = async (cowId) => {

    db = await getDatabase();


    const res = await db.runAsync(
        `UPDATE Cow SET exitDate = CURRENT_DATE WHERE id = ?`,
        [cowId]
    );

    return res;
};

export const getAllCowsExitedByUserIdAndPhaseAndGender = async (userId, phase, gender) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL AND phase = ? AND gender = ?`,
        [userId, phase, gender]
    );

    return res;
};

export const getAllCowsExitedByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL`,
        [userId]
    );

    return res;
};

export const getAllCalvesExitedByUserId = async (userId) => {

    db = await getDatabase();

    const res = await db.getAllAsync(
        `SELECT * FROM Cow WHERE user_fk = ? AND exitDate is NOT NULL AND phase = 'Ternero'`,
        [userId]
    );

    return res;
};