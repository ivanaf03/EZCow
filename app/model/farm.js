import { getDatabase } from "./bd";

export const insertFarmUbication = async (userId, latitude, longitude) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `UPDATE User SET latitude = ?, longitude = ? WHERE id = ?`,
      [latitude, longitude, userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getFarmUbicationByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(
      `SELECT latitude, longitude FROM User WHERE id = ?`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
