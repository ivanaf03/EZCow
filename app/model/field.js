import { getDatabase } from "./bd";

export const insertField = async (name, latitude, longitude, user_fk) => {
  const db = await getDatabase();
  try {
    await db.runAsync(
      `INSERT INTO Field (name, latitude, longitude, user_fk) VALUES (?, ?, ?, ?)`,
      [name, latitude, longitude, user_fk]
    );
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllFieldsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(`SELECT * FROM Field WHERE user_fk = ?`, [
      userId,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
