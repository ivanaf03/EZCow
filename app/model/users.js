import CryptoJS from "crypto-js";
import { getDatabase } from "./bd";

const hashPassword = (password) => {
  return CryptoJS.SHA256(password).toString();
};

export const insertUser = async (name, email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `INSERT INTO User (name, email, password) VALUES (?, ?, ?)`,
      [name, email, hashedPassword]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const insertUserGoogle = async (name, email, google_id) => {
  const db = await getDatabase();
  try {
    const existingUser = await db.getFirstAsync(
      `SELECT * FROM User WHERE email = ?`,
      [email]
    );
    if (existingUser) {
      return existingUser;
    }
    const res = await db.runAsync(
      `INSERT INTO User (name, email, google_id) VALUES (?, ?, ?)`,
      [name, email, google_id]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getUserByEmail = async (email) => {
  const db = await getDatabase();
  try {
    const res = await db.getFirstAsync(`SELECT * FROM User WHERE email = ?`, [
      email,
    ]);
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const changePassword = async (email, password) => {
  const db = await getDatabase();
  const hashedPassword = hashPassword(password);
  try {
    const res = await db.runAsync(
      `UPDATE User SET password = ? WHERE email = ?`,
      [hashedPassword, email]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};
