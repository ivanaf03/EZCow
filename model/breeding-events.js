import { getDatabase } from "../model/bd";

export const insertBreedingEvent = async (
  cow_fk,
  eventName,
  description,
  date
) => {
  const db = await getDatabase();
  try {
    const res = await db.runAsync(
      `INSERT INTO BreedingEvent (cow_fk, eventName, description, date) VALUES (?, ?, ?, ?)`,
      [cow_fk, eventName, description, date]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};

export const getAllBreedingEventsByUserId = async (userId) => {
  const db = await getDatabase();
  try {
    const res = await db.getAllAsync(
      `SELECT * FROM BreedingEvent JOIN Cow ON BreedingEvent.cow_fk = Cow.id WHERE user_fk = ? and exitDate IS NULL`,
      [userId]
    );
    return res;
  } catch (error) {
    console.log(error);
    return null;
  }
};