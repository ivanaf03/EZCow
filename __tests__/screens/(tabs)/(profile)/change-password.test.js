import React from "react";

import { render, waitFor, fireEvent } from "@testing-library/react-native";
import { router } from "expo-router";
import { Alert } from "react-native";

import ChangePassword from "../../../../app/(tabs)/(profile)/change-password";
import { getUserByEmail, changePassword } from "../../../../app/model/users";

jest.mock("../../../../store/user-provider", () => ({
  useUser: jest.fn(() => ({
    user: {
      id: "1",
      name: "TestUser",
      email: "test@test.com",
      password: "12345678",
    },
    login: jest.fn(),
  })),
}));

jest.mock("expo-router", () => ({
  router: {
    replace: jest.fn(),
    push: jest.fn(),
  },
}));

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: () => null,
}));

jest.mock("../../../../app/model/users", () => ({
  getUserByEmail: jest.fn(),
  changePassword: jest.fn(),
}));

jest.spyOn(Alert, "alert");

describe("ChangePassword", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    const tree = render(<ChangePassword />);
    expect(tree).toMatchSnapshot();
  });

  it("should change password if actual password is correct", async () => {
    changePassword.mockResolvedValueOnce(true);
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "123456789");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "123456789");
    getUserByEmail.mockResolvedValue({
      email: "test@example.com",
      password: "12345678",
    });
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    await waitFor(() => expect(router.replace).toHaveBeenCalledWith("profile"));
  });

  it("should show an error if user is logged with google", async () => {
    const useUserMock =
      require("../../../../store/user-provider").useUser;
    useUserMock.mockReturnValue({
      user: {
        id: "1",
        name: "TestUser",
        email: "test@test.com",
        password: null,
      },
    });
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "123456789");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "123456789");
    getUserByEmail.mockResolvedValue({
      email: "test@example.com",
      password: null,
    });
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Solo podrás cambiar la contraseña si no has iniciado sesión con Google."
    );
  });

  it("should show an error if actual password is empty", async () => {
    const useUserMock =
      require("../../../../store/user-provider").useUser;
    useUserMock.mockReturnValue({
      user: {
        id: "1",
        name: "TestUser",
        email: "test@test.com",
        password: "12345678",
      },
    });
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "123456789");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "123456789");
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if actual password is not the same as the user password", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345656");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "123456789");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "123456789");
    getUserByEmail.mockResolvedValue({
      email: "test@example.com",
      password: "123456789",
    });
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "La contraseña actual no es correcta."
    );
  });

  it("should show an error if new password is empty", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "123456789");
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if confirm new password is empty", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "123456789");
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if new password is different from confirm new password", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "123456789");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "12345678");
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Las contraseñas no coinciden."
    );
  });

  it("should show an error if new password is less than 8 characters", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "1234567");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "1234567");
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "La contraseña debe tener al menos 8 caracteres."
    );
  });

  it("should show an error if actual password is incorrect", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "12345678");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "123456789");
    getUserByEmail.mockResolvedValue({
      email: "test@example.com",
      password: "12345678",
    });
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    waitFor(() =>
      expect(Alert.alert).toHaveBeenCalledWith(
        "Error",
        "La contraseña actual no es correcta."
      )
    );
  });

  it("should show an error if new password is the same as the actual", async () => {
    const { getByPlaceholderText, getByTestId } = render(<ChangePassword />);
    const actualPasswordInput = getByPlaceholderText("Contraseña actual");
    fireEvent.changeText(actualPasswordInput, "12345678");
    const newPasswordInput = getByPlaceholderText("Nueva contraseña");
    fireEvent.changeText(newPasswordInput, "12345678");
    const confirmNewPasswordInput = getByPlaceholderText(
      "Confirmar nueva contraseña"
    );
    fireEvent.changeText(confirmNewPasswordInput, "12345678");
    getUserByEmail.mockResolvedValue({
      email: "test@example.com",
      password: "12345678",
    });
    const changePasswordButton = getByTestId("change-password-button");
    fireEvent.press(changePasswordButton);
    waitFor(() =>
      expect(Alert.alert).toHaveBeenCalledWith(
        "Error",
        "La nueva contraseña debe ser diferente a la actual."
      )
    );
  });
});
