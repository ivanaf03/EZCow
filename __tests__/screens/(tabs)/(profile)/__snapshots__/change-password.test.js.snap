// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword should render correctly 1`] = `
<RCTScrollView
  style={
    [
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <View>
    <Text
      style={
        [
          {
            "marginTop": 16,
          },
          {
            "marginLeft": 16,
            "marginRight": 16,
          },
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 36,
            "lineHeight": 40,
          },
          {
            "fontFamily": "Nunito-Bold",
          },
          {
            "borderBottomWidth": 2,
          },
          {
            "borderBottomColor": "#ffffff",
            "borderLeftColor": "#ffffff",
            "borderRightColor": "#ffffff",
            "borderTopColor": "#ffffff",
          },
        ]
      }
    >
      Cambiar contraseña
    </Text>
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "alignItems": "center",
          },
          {
            "marginTop": 32,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "width": "90%",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "backgroundColor": "#272727",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
              {
                "borderBottomWidth": 2,
                "borderLeftWidth": 2,
                "borderRightWidth": 2,
                "borderTopWidth": 2,
              },
              {
                "borderBottomColor": "#ffffff",
                "borderLeftColor": "#ffffff",
                "borderRightColor": "#ffffff",
                "borderTopColor": "#ffffff",
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "flexDirection": "row",
                },
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  },
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 18,
                    "lineHeight": 28,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                  },
                  [
                    {
                      "marginLeft": 16,
                    },
                  ],
                ]
              }
            >
              Atención! Solo podrás cambiar la contraseña si no has iniciado sesión con Google
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 32,
          },
          {
            "paddingLeft": 16,
            "paddingRight": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
            },
          ]
        }
      >
        <View>
          <View
            style={
              [
                {
                  "marginTop": -4,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              Contraseña actual
              :
            </Text>
            <TextInput
              onChangeText={[Function]}
              placeholder="Contraseña actual"
              style={
                [
                  {
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  },
                  {
                    "backgroundColor": "#ffffff",
                  },
                  {
                    "borderBottomWidth": 2,
                    "borderLeftWidth": 2,
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "borderBottomColor": "#2254ba",
                    "borderLeftColor": "#2254ba",
                    "borderRightColor": "#2254ba",
                    "borderTopColor": "#2254ba",
                  },
                  {
                    "borderBottomLeftRadius": 16,
                    "borderBottomRightRadius": 16,
                    "borderTopLeftRadius": 16,
                    "borderTopRightRadius": 16,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "marginTop": -4,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Light",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              Nueva contraseña
              :
            </Text>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Nueva contraseña"
                secureTextEntry={true}
                style={
                  [
                    {
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                    },
                    {
                      "marginRight": 8,
                    },
                    {
                      "backgroundColor": "#ffffff",
                    },
                    {
                      "borderBottomWidth": 2,
                      "borderLeftWidth": 2,
                      "borderRightWidth": 2,
                      "borderTopWidth": 2,
                    },
                    {
                      "borderBottomColor": "#2254ba",
                      "borderLeftColor": "#2254ba",
                      "borderRightColor": "#2254ba",
                      "borderTopColor": "#2254ba",
                    },
                    {
                      "borderBottomLeftRadius": 16,
                      "borderBottomRightRadius": 16,
                      "borderTopLeftRadius": 16,
                      "borderTopRightRadius": 16,
                    },
                    {
                      "flexBasis": "0%",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    {
                      "fontFamily": "Nunito-Medium",
                    },
                  ]
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                testID="eye-icon"
              />
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "marginTop": -4,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Light",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              Confirmar nueva contraseña
              :
            </Text>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Confirmar nueva contraseña"
                secureTextEntry={true}
                style={
                  [
                    {
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                    },
                    {
                      "marginRight": 8,
                    },
                    {
                      "backgroundColor": "#ffffff",
                    },
                    {
                      "borderBottomWidth": 2,
                      "borderLeftWidth": 2,
                      "borderRightWidth": 2,
                      "borderTopWidth": 2,
                    },
                    {
                      "borderBottomColor": "#2254ba",
                      "borderLeftColor": "#2254ba",
                      "borderRightColor": "#2254ba",
                      "borderTopColor": "#2254ba",
                    },
                    {
                      "borderBottomLeftRadius": 16,
                      "borderBottomRightRadius": 16,
                      "borderTopLeftRadius": 16,
                      "borderTopRightRadius": 16,
                    },
                    {
                      "flexBasis": "0%",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    {
                      "fontFamily": "Nunito-Medium",
                    },
                  ]
                }
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                testID="eye-icon"
              />
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "display": "flex",
                },
                {
                  "flexDirection": "row",
                },
                {
                  "justifyContent": "space-between",
                },
                {
                  "alignItems": "center",
                },
                {
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
                {
                  "backgroundColor": "#272727",
                },
                {
                  "borderBottomWidth": 2,
                  "borderLeftWidth": 2,
                  "borderRightWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#6bc0eb",
                  "borderLeftColor": "#6bc0eb",
                  "borderRightColor": "#6bc0eb",
                  "borderTopColor": "#6bc0eb",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
              ]
            }
            testID="change-password-button"
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Nunito-SemiBold",
                  },
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 20,
                    "lineHeight": 28,
                  },
                  {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Cambiar contraseña
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
